File: Code/bramex.R
## bramex data set ##

# Importing data #
df = read.csv('https://raw.githubusercontent.com/mcanela-iese/Stats_Course/master/Data/bramex.csv')
str(df)
head(df)
tail(df)
summary(df)

# Regression line #
regmod = lm(formula=mex ~ bra, data=df)
names(regmod)
regmod$coefficients
summary(regmod)
summary(regmod)$r.squared

# Correlation analysis #
cor(df$bra, df$mex)
cor(df[, -1])
cov(df[, -1])

# Figure 1 (customize) #
pdf('Figures/fig 1.1.pdf', width=3.25, height=3.5, pointsize=7)
plot(bramex$bra, bramex$mex, main='', xlab='Brazil', ylab='Mexico', pch=20)
dev.off()

# Figure 2 (customize) #
pdf('Figures/fig 1.2.pdf', width=3.25, height=3.5, pointsize=7)
plot(df$bra, df$mex, main='', xlab='Brazil', ylab='Mexico', pch=20)
abline(regmod)
dev.off()

File: Code/bramex.py
## bramex data set ##

# Importing data #
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/' + \
    'mcanela-iese/Stats_Course/master/Data/bramex.csv')
df.info()
df.head()
df.tail()
df.describe()

# Regression line #
import statsmodels.formula.api as smf
regmod = smf.ols(formula = 'mex ~ bra', data = df).fit()
regmod.summary()

# Correlation analysis #
df[['bra', 'mex']].corr()

# Figure 1 (customize) #
import matplotlib.pyplot as plt
plt.scatter(df['bra'], df['mex'], color='0.3', s=16)
plt.xlabel('Brazil')
plt.ylabel('Mexico')
# plt.savefig('Figures/fig 1.1.pdf')
plt.show()

# Figure 2 (customize) #
import statsmodels.api as sm
sm.graphics.abline_plot(model_results=regmod, color='0.3')
plt.scatter(df['bra'], df['mex'], color='0.3', s=16)
plt.xlabel('Brazil')
plt.ylabel('Mexico')
# plt.savefig('Figures/fig 1.1.pdf')
plt.show()
